### ISSUE TITLE
b-table provider function not being called

### ISSUE THREAD

**Issue by jmdevep** [bootstrap-vue/bootstrap-vue#2015]
Hello,

I'm using a provider items function to provide the data array, but it's not being called besides having busy prop. set to false.

This is my b-table

                       `<b-table id="my-table"
                                    ref="table"
                                    show-empty
                                    :items="myProvider"
                                    :fields="fields"
                                    :current-page="currentPage"
                                    :per-page="perPage"
                                    :busy="isBusy"
                                    :filter="filter"
                            >`
My data
          `data(){
            return{
                isBusy: false,
                fields: [
                    { key: 'inicioEvento', label: 'Inicio', sortable: true, sortDirection: 'desc' },
                    { key: 'servicio.nombre', label: 'Servicio', sortable: true, sortDirection: 'desc' },  
                    { key: 'persona.nombre', label: 'Persona' },
                    { key: 'estado.nombre', label: 'Estado' },
                    { key: 'direccion', label: 'Direccion' },
                    { key: 'actions', label: 'Acciones' }
                ],
                currentPage: 1,
                perPage: 5,
                totalRows: 1,
                pageOptions: [ 5, 10, 15 ],
                sortBy: null,
                sortDesc: false,
                filter: null,
                modalInfo: { title: '', content: '' }
                }
        },`

And the provider function

             `myProvider (ctx) {
                console.log("Proveedor datos");
                this.isBusy = true;
                
                return axios.get(`${process.env.BASE_URL}/api/evento/lista-eventos`, {
                    params: {
                        condiciones: {
                            orden: 'DESC',
                            tamanoPagina: ctx.perPage,
                            indicePagina: ctx.currentPage,
                            campo: 'nombre',
                        },
                    }
                })
                    .then((res)=>{
                        console.log(res);
                        if(res.data.resultado == 100){
                            let items = res.data.listaEventos;
                         
                             this.isBusy = false;   
                             return items;       
                        }
                }).catch(error => {
                    this.isBusy = false;        
                    return [];
                })
                return null;
            }   `

Note that I also tried setting :busy.sync, and not handling busy manually at all, and all no-provider-* to true. I tried both async (with or without callback), and sync methods for the myProvider function and different combinations of b-table properties. I even show my busy property on screen to make sure it's set to false.

Version:
v2.0.0-rc.11
On Vue-cli 
"vue": "^2.5.2",
"bootstrap-vue": "^2.0.0-rc.11",

**Comment by kl3sk** [comment]
What is the purpose of your `return null;` at the end of your provider ?
Does your items match your field structure ?

Your `console.log` inside your `axios.then` does return something ?

Indeed if your `if` is false your then never return something, a `then` should always return something, at least an empty array `[]`


May it help

**Comment by jmdevep** [comment]
Yes! it was indeed the return null statement, thanks for the clarification and the further info on return from SO.
